src/Stoa.ts

```diff
@@ -1425,6 +1432,50 @@ class Stoa extends WebService
                 }
             })
     }
+
+    /**
+     * GET /averagefeechart/
+     *@param req
+     *@param res
+     * Called when a request is received through the `/averagefeechart/` handler
+     *
+     * @returns Returns average transaction fee from given req
+     */
+    private async averageFeeChart(req: express.Request, res: express.Response){        
+        let to = await Time.msToTime(Date.now());
+        let from = await JSBI.subtract(JSBI.BigInt(to.seconds), JSBI.BigInt(req.params.from));
+        let num = Number(from.toString());
+        logger.http(`GET /averagefeechart/${req.params.from}`);
+        
+        this.ledger_storage.calculateAvgFeeChart(Number(from.toString()),to.seconds).then((data: any) => {
+            if ((data === undefined)) {
+                res.status(500).send("Failed to data lookup");
+                return;
+              }
+           else if (data.length === 0) {
+               return res.status(204).send(`The data does not exist.`);
+            }
+            else {
+                let avgFeelist: Array<IAvgFee> = [];
+                for (const row of data) {
+                    avgFeelist.push(
+                        {
+                            height: row.height,
+                            time_stamp: row.time_stamp,
+                            average_tx_fee: row.average_tx_fee,
+                            total_tx_fee: row.total_tx_fee,
+                            total_payload_fee: row.total_payload_fee,
+                            total_fee: row.total_fee
+                        }
+                    )
+                }
+            return res.status(200).send(JSON.stringify(avgFeelist));
```

Comment:
How many data do you make and return if you look up the 5 year chart on the screen?
Depending on the conditions on the screen, there is a need to adjust the data resolution flexibly.